src/main/java/de/tum/in/www1/artemis/service/user/UserService.java

```diff
@@ -582,4 +582,57 @@ private void removeUserFromGroupInternal(User user, String group) {
             saveUser(user);
         }
     }
+
+    /**
+     * This method first tries to find the student in the internal Artemis user database (because the user is most probably already using Artemis).
+     * In case the user cannot be found, we additionally search the (TUM) LDAP in case it is configured properly.
+     *
+     *       @param registrationNumber     the registration number of the user
+     *       @param courseGroupName        the courseGroup the user has to be added to
+     *       @param courseGroupRole        the courseGroupRole enum
+     *       @param login                  the login of the user
+     *       @return the found student, otherwise returns an emtpy optional
+     *
+     * */
+    public Optional<User> findUserAndAddToCourse(String registrationNumber, String courseGroupName, Role courseGroupRole, String login) {
```

Comment:
Why not to have two different functions. One is `findUser` and the other one `addUserToCourse`?

Reply:
We actually have two different methods (this one just finds the student while this.addUserToGroup(...)  adds to the course, however i wanted to be precise about the functionality)